plugins/jobs/web_client/views/JobListWidget.js

```diff
@@ -24,21 +24,28 @@ var JobListWidget = View.extend({
 
     initialize: function (settings) {
         var currentUser = getCurrentUser();
+        this.showAllJobs = !!settings.allJobsMode;
         this.columns = settings.columns || this.columnEnum.COLUMN_ALL;
         this.filter = settings.filter || {
-            userId: currentUser.id
+            userId: currentUser ? currentUser.id : null
         };
         this.typeFilter = {};
         this.statusFilter = {};
 
         this.collection = new JobCollection();
+        if (this.showAllJobs) {
+            this.collection.resourceName = 'job/all';
+        }
         this.collection.sortField = settings.sortField || 'created';
         this.collection.sortDir = settings.sortDir || SORT_DESC;
         this.collection.pageLimit = settings.pageLimit || this.collection.pageLimit;
 
         this.collection.on('g:changed', function () {
+            this.typeFilter = {};
+            this.statusFilter = {};
```

Comment:
It would be better to not reset the filters on a page change.  Just commenting out these two `= {}` lines seems to behave  more like I'd expect.  If a type or status isn't on the new page, it will be reenabled when you navigate back (it would ideally remember the setting, but that doesn't have to change in this PR).